core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts

```diff
@@ -46,29 +48,54 @@ export class DotFavoritePageComponent implements OnInit, AfterViewInit, OnDestro
-            isAdmin: this.config.data.page.pageState.user.admin,
-            imgWidth: this.config.data.page.pageState.viewAs.device?.cssWidth,
-            imgHeight: this.config.data.page.pageState.viewAs.device?.cssHeight,
-            pageRenderedHtml: this.config.data.page.pageRenderedHtml
+            favoritePageUrl: this.config.data.page.favoritePageUrl,
+            favoritePage: this.config.data.page.favoritePage
         });
     }
 
     ngOnInit(): void {
-        const { page }: { page: DotFavoritePageProps } = this.config.data;
-
-        this.form = this.fb.group(this.setupForm(page.pageState));
-        this.isFormValid$ = this.form.valueChanges.pipe(
-            takeUntil(this.destroy$),
-            map(() => {
-                return this.form.valid;
-            }),
-            startWith(false)
-        );
-
-        this.store.currentUserRoleId$
+        // Needed to avoid browser to cache thumbnail img when reloaded, since it's fetched from the same URL
+        this.timeStamp = new Date().getTime().toString();
+
+        this.store.formState$
             .pipe(takeUntil(this.destroy$))
-            .subscribe((currentUserRoleId: string) => {
-                this.form.get('currentUserRoleId').setValue(currentUserRoleId);
+            .subscribe((formStateData: DotFavoritePageFormData) => {
+                if (formStateData) {
+                    this.form = this.fb.group({
+                        currentUserRoleId: [formStateData?.currentUserRoleId, Validators.required],
+                        inode: [formStateData?.inode],
+                        thumbnail: [formStateData?.thumbnail, Validators.required],
+                        title: [formStateData?.title, Validators.required],
+                        url: [{ value: formStateData?.url, disabled: true }, Validators.required],
+                        order: [formStateData?.order, Validators.required],
+                        permissions: [formStateData?.permissions]
+                    });
+                    this.isFormValid$ = this.form.valueChanges.pipe(
+                        takeUntil(this.destroy$),
+                        map(() => {
+                            return this.form.valid;
+                        }),
+                        startWith(false)
+                    );
+
+                    requestAnimationFrame(() => {
+                        if (formStateData?.inode) {
+                            this.form.get('thumbnail').setValue(formStateData?.thumbnail);
+                        } else {
+                            this.setPreviewThumbnailListener();
+                        }
+                    });
+                }
+            });
+
+        this.store.renderThumbnail$
+            .pipe(takeUntil(this.destroy$))
+            .subscribe((renderThumbnail: boolean) => {
+                if (renderThumbnail) {
+                    requestAnimationFrame(() => {
+                        this.setPreviewThumbnailListener();
```

Comment:
Every time we call `renderThumbnail` we do a `addEventListener` which looks like is multiple times during the lifecycle, that could be a memory leak.

Why don't you wrap `dot-html-to-image` into an angular component that implement controlvalueaccessor and you can use normally inside the form?